### PR TITLE
refactor(material-experimental/mdc-list): no longer use deprecated styles

### PR DESCRIPTION
**Pull request by devversion**
We recently switched the MDC list to deprecated styles as the MDC team
started reworking the MDC list and our usage is not compatible.

With this change, we start using the standard non-deprecated MDC list
styles. This allows us to simplify a few custom CSS overrides due the
new MDC list API being more aligned with the API we want to support for
backwards compatibility with the non-MDC list (e.g. per list-item
icons and avatars instead of it being a per-list setting).

### CODE REVIEW

**Comment by devversion** [comment]
Blocked on https://github.com/material-components/material-components-web/pull/7021

File: src/dev-app/list/list-demo.ts [hunk 1]
```diff
@@ -35,13 +35,13 @@ export class ListDemo {
...
       message: 'Did you want to go on Sunday? I was thinking that might work.',
-      image: 'https://angular.io/generated/images/bios/julie-ralph.jpg'
+      image: 'https://angular.io/generated/images/bios/cindygreenekaplan.jpg'
     },
     {
       from: 'Mary',
       subject: 'Summer BBQ',
       message: 'Wish I could come, but I have some prior obligations.',
-      image: 'https://angular.io/generated/images/bios/juleskremer.jpg'
+      image: 'https://angular.io/generated/images/bios/twerske.jpg'
```
Comment (devversion) [inline src/dev-app/list/list-demo.ts:44]: @cindygk @twerske Are you both fine being in this team-internal demo site? The current images no longer exist.

**Comment by devversion** [comment]
This is now ready for review. The MDC fixes @crisbeto and I landed are available now.

<sup>I've been waiting for #22504 to land; but given it not being merge safe, I'm just updating MDC canary here too</sup>

**Review by mmalerba — APPROVED** [review]
LGTM

**Review by jelbourn — APPROVED** [review]
lgtm

**Pull request merged**